lib/msal-common/src/network/ThrottlingManager.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { INetworkModule, NetworkRequestOptions } from "./INetworkModule";
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { Constants, HeaderNames } from "../utils/Constants";
+import { RequestThumbprint } from "./RequestThumbprint";
+import { RequestThumbprintValue } from "./RequestThumbprintValue";
+import { AuthError } from "../error/AuthError";
+
+export type NetworkResponse<T> = {
+    headers: Map<string, string>;
+    body: T;
+    status: number;
+};
+
+export abstract class ThrottlingManager {
+    protected networkClient: INetworkModule;
+    abstract getThrottlingItem(thumbprint: RequestThumbprint): RequestThumbprintValue | null;
+    abstract setThrottlingItem(thumbprint: RequestThumbprint, thumbprintValue: RequestThumbprintValue): void;
+    abstract removeThrottlingItem(thumbprint: RequestThumbprint): boolean;
+
+    constructor(networkClient: INetworkModule) {
+        this.networkClient = networkClient;
+    }
+
+    public async sendPostRequest<T>(thumbprint: RequestThumbprint, tokenEndpoint: string, options: NetworkRequestOptions): Promise<NetworkResponse<T>> {
+
+        this.preProcess(thumbprint);
+        const response = await this.networkClient.sendPostRequestAsync<T>(tokenEndpoint, options);
+        this.postProcess(thumbprint, response);
+
+        return response;
+    }
+
+    public preProcess(thumbprint: RequestThumbprint): void {
```

Comment:
Ideally, we could make `preProcess` and `postProcess` static to allow other libraries to consume them without passing arround a `ThrottlingManager`, but I couldn't figure out a good way of accomplishing that as we also need to call the abstract functions here. Any thoughts?

Reply:
I don't think you need to make this static. You could make them protected.

Reply:
I'm still unclear on why this needs to be abstract. Each library should provide their own `cacheManager` which you can use without needing each library to implement their own `ThrottlingManager`.

Reply:
Since they are public or protected sub-classes can implement them and overwrite functionality

Reply:
I think what Thomas is saying is that we don't necessarily need this to be abstract if we just directly use the provided cacheManager instead of forcing each library to implement it's own storage relative to throttling. cacheManager should be generic. Node just isn't set up to do that today, it would need to be incremental.

Reply:
Ended up moving these into a separate file, no longer have any abstract functions and we're directly using `cacheManager`.